cli/sync_service/download.go

```diff
@@ -157,7 +168,56 @@ func VerifyDataSig(dataReader io.Reader, publicKey string, signature string, has
 				return false, err
 			}
 
+			// rename the .tmp file
+			if err := os.Rename(tmpFileName, fileName); err != nil {
+				return false, err
+			}
+
 			return true, nil
 		}
 	}
 }
+
+func writeDateStreamToFile(dataReader io.Reader, fileName string) error {
+	file, err := os.OpenFile(fileName, os.O_WRONLY|os.O_CREATE, 0600)
+	if err != nil {
+		return err
+	}
+	defer file.Close()
+
+	wc := &WriteCounter{
+		PassInWriter: file,
+		ChunkNumber:  1,
+	}
+
+	if _, err := file.Seek(0, io.SeekStart); err != nil {
+		return err
+	}
+
+	if _, err := io.Copy(wc, dataReader); err != nil && err != io.EOF {
+		return err
+	}
+	fmt.Println()
+
+	return nil
+}
+type WriteCounter struct {
+	PassInWriter io.Writer
+	ChunkNumber int
+}
+
+func (wc *WriteCounter) Write(p []byte) (n int, err error) {
+	n, err = wc.PassInWriter.Write(p)
+	wc.ChunkNumber += 1
+	wc.PrintProgress(n)
+	return n, err
+}
+
+func (wc *WriteCounter) PrintProgress(n int) {
```

Comment:
I really dont like how the dots work. It would look better if we could use the % counter like we do for upload.